### PR TITLE
[Android] Implement the feature about data list.

### PR DESCRIPTION
**Pull request by wuhengzhi**
This patch is to implement the feature about data list.
Most of the code was ported from android_webview.

BUG=XWALK-1457

### CODE REVIEW

**Comment by wuhengzhi** [comment]
@junmin-zhu @lincsoon Please help to review, thanks.

**Comment by lincsoon** [comment]
ping @axinging @fujunwei

**Comment by baleboy** [comment]
what's the way forward with this PR? It's been in review for several months.

**Comment by wuhengzhi** [comment]
I will rebase the commits and ping the reviewers.

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkAutofillClient.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+/**
+ * Java counterpart to the XWalkAutofillClient. This class is owned by XWalkContent and has
+ * a weak reference from native side.
+ */
+@JNINamespace("xwalk")
+public class XWalkAutofillClient {
+
+    private final long mNativeXWalkAutofillClient;
+    private AutofillPopup mAutofillPopup;
+    private ViewGroup mContainerView;
```
Comment (axinging) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkAutofillClient.java]: mContainerView wasn't used?

**Comment by RNCRyan** [comment]
+1 on the need for this.  Thanks!

**Comment by wuhengzhi** [comment]
@lincsoon @axinging @halton Please help to review, thanks.

**Comment by halton** [comment]
@lincsoon I'll reply on your review.

**Comment by wuhengzhi** [comment]
ping @axinging @lincsoon @fujunwei thanks.

**Comment by lincsoon** [comment]
LGTM. Please rebase so that I can merge

**Comment by wuhengzhi** [comment]
@lincsoon Rebase is done.

**Pull request merged**